createDocu.groovy

```diff
@@ -0,0 +1,359 @@
+import groovy.io.FileType;
+import org.yaml.snakeyaml.Yaml
+import org.codehaus.groovy.control.CompilerConfiguration
+import com.sap.piper.DefaultValueCache
+import java.util.regex.Matcher
+
+roots = [
+    'vars',
+    'src',
+    ]
+
+def stepsDir
+def outDir
+def stepDocuDir
+
+//
+// assign parameters
+
+if(args.length >= 1)
+  stepsDir = new File(args[0])
+
+stepsDir = stepsDir ?: new File('vars')
+
+if(args.length >= 2)
+  outDir = new File(args[1])
+
+outDir = outDir ?: new File('out')
+
+if(args.length >= 3)
+  stepsDocuDir = new File(args[2])
+
+stepsDocuDir = stepsDocuDir ?: new File('documentation/docs/steps')
+
+// assign parameters
+//
+
+//
+// sanity checks
+
+if( ! outDir.exists() ) {
+  if(! outDir.mkdirs()) {
+    System.err << "Cannot create output direcrory '${outDir}'.\n"
+    System.exit(1)
+  }
+}
+
+if( !stepsDocuDir.exists() ) {
+  System.err << "Steps docu dir '${stepsDocuDir}' does not exist.\n"
+  System.exit(1)
+}
+
+if( !stepsDir.exists() ) {
+  System.err << "Steps dir '${stepsDir}' does not exist.\n"
+  System.exit(1)
+}
+
+// sanity checks
+//
+
+def blacklist = [
```

Comment:
to have a better overview, we should add all steps here and comment the ones to generate the docs for.

Reply:
I think we should rely on the list of steps available in the directory, thus no need to add something. Default should be to generate the documentation.
Only if a documentation is not necessary/... action should be required.

Reply:
Blacklist was added simply in order to focus on some example steps during rapid development. IMO we should not have a blacklist here in the long run. It is my intention to remove the blacklist.

In case we are not able to generate the step docu for some steps we should implement a `@no-generated-docu` annotation (maybe only a pseudo-annotation inside a javadoc comment).

In case we go for generating the documentation manually (... what is IMO fine at the beginning, in the long run we should IMO generate the docu as part of the build process) we could also think about generating the docu only for dedicated steps provided as parameter at the `createDocu.sh` invocation.